lib/std/crypto/Certificate.zig

```diff
@@ -1124,45 +1238,88 @@ pub const rsa = struct {
             // unlikely that exponents larger than 32 bits are being used for anything
             // Windows commonly does.
             // [1] https://learn.microsoft.com/en-us/windows/win32/api/wincrypt/ns-wincrypt-rsapubkey
-            if (pub_bytes.len > 4) return error.CertificatePublicKeyInvalid;
-            const _e = Fe.fromBytes(_n, pub_bytes, .big) catch return error.CertificatePublicKeyInvalid;
+            if (exp.len > 4) return error.CertificatePublicKeyInvalid;
+            const _e = Fe.fromBytes(_n, exp, .big) catch return error.CertificatePublicKeyInvalid;
             if (!_e.isOdd()) return error.CertificatePublicKeyInvalid;
             const e_v = _e.toPrimitive(u32) catch return error.CertificatePublicKeyInvalid;
             if (e_v < 2) return error.CertificatePublicKeyInvalid;
 
-            return .{
-                .n = _n,
-                .e = _e,
-            };
+            return .{ .n = _n, .e = _e };
         }
 
         // RFC8017 Appendix A.1.1
-        pub fn fromDer(pub_key: []const u8) !PublicKey {
-            const pub_key_seq = try der.Element.parse(pub_key, 0);
-            if (pub_key_seq.identifier.tag != .sequence) return error.CertificateFieldHasWrongDataType;
-            const modulus_elem = try der.Element.parse(pub_key, pub_key_seq.slice.start);
+        pub fn fromDer(bytes: []const u8) !PublicKey {
+            const seq = try der.Element.parse(bytes, 0);
+            if (seq.identifier.tag != .sequence) return error.CertificateFieldHasWrongDataType;
+            const modulus_elem = try der.Element.parse(bytes, seq.slice.start);
             if (modulus_elem.identifier.tag != .integer) return error.CertificateFieldHasWrongDataType;
-            const exponent_elem = try der.Element.parse(pub_key, modulus_elem.slice.end);
+            const exponent_elem = try der.Element.parse(bytes, modulus_elem.slice.end);
             if (exponent_elem.identifier.tag != .integer) return error.CertificateFieldHasWrongDataType;
             // Skip over meaningless zeroes in the modulus.
-            const modulus_raw = pub_key[modulus_elem.slice.start..modulus_elem.slice.end];
-            const modulus_offset = for (modulus_raw, 0..) |byte, i| {
-                if (byte != 0) break i;
-            } else modulus_raw.len;
-            const exponent = pub_key[exponent_elem.slice.start..exponent_elem.slice.end];
+            const modulus_raw = bytes[modulus_elem.slice.start..modulus_elem.slice.end];
+            const modulus_offset = std.mem.indexOfNone(u8, modulus_raw, &[_]u8{0}) orelse modulus_raw.len;
             const modulus = modulus_raw[modulus_offset..];
+            const exponent = bytes[exponent_elem.slice.start..exponent_elem.slice.end];
 
-            return try fromBytes(exponent, modulus);
+            return try fromBytes(modulus, exponent);
```

Comment:
Why is the order of modulus and exponent reversed?

Reply:
After mixing them up, I changed the order of the args to match the DER serialization order and struct member order.

Reply:
If they are easy to mix up, why not pass these parameters as a struct instead?

Reply:
Eh, I don't think 2 members need a struct. I think they just need a consistent order everywhere.

This will be moved to a separate crypto PR soon.